onnxruntime/core/optimizer/gather_internal_replacement.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+// gather_internal_replacement.j
+
+#pragma once
+
+#include "core/optimizer/rewrite_rule.h"
+
+namespace onnxruntime {
+
+/**
+@Class GatherInternalReplacement
+
+Rewrite rule that replaces the GatherInternal with Gather when the extra
+outputs computed by GatherInternal are not needed. This happens
+when the GatherInternal node is not connected to a GatherGrad node.
+
+*/
+class GatherInternalReplacement : public RewriteRule {
```

Comment:
Could we add more information to this class? Some ASCII graphs may be quite helpful like [this](https://github.com/microsoft/onnxruntime/blob/f6edf135132bbcbeb8e2294078d305c4353422cc/onnxruntime/core/optimizer/gemm_sum_fusion.h#L24).